mypy/meet.py

```diff
@@ -137,6 +139,21 @@ def get_possible_variants(typ: Type) -> List[Type]:
         return [typ]
 
 
+def is_enum_overlapping_union(x: ProperType, y: ProperType) -> bool:
+    return (
+        isinstance(x, Instance) and x.type.is_enum and
+        isinstance(y, UnionType) and
+        all(x.type == p.fallback.type
+            for p in (get_proper_type(z) for z in y.relevant_items())
+            if isinstance(p, LiteralType))
```

Comment:
```suggestion
        all(isinstance(p, LiteralType)) and x.type == p.fallback.type
            for p in (get_proper_type(z) for z in y.relevant_items()))
```
Should it be this instead? As I understand it, this function checks that `x` is an enum and `y` is a union containing only members of `x`.

It would be helpful to add a docstring to this function.

Reply:
Upon looking at this again, I think the previous implementation was wrong since we only care about partial overlap. Fixed and documented accordingly.